src/Microsoft.Health.Dicom.Functions.Client/DicomAzureFunctionsClient.cs

```diff
@@ -66,63 +67,71 @@ internal class DicomAzureFunctionsClient : IDicomOperationsClient
-    {
-        cancellationToken.ThrowIfCancellationRequested();
-
-        // TODO: Pass token when supported
-        DurableOrchestrationStatus state = await GetStatusInternalAsync(operationId);
-
-        if (state == null)
-        {
-            return null;
-        }
-
-        OperationStatus status = state.RuntimeStatus.ToOperationStatus();
-        DicomOperation type = state.GetDicomOperation();
-        IOperationCheckpoint checkpoint = ParseCheckpoint(type, state);
-        return new OperationState<DicomOperation>
-        {
-            CreatedTime = checkpoint.CreatedTime ?? state.CreatedTime,
-            LastUpdatedTime = state.LastUpdatedTime,
-            OperationId = operationId,
-            PercentComplete = checkpoint.PercentComplete.HasValue && status == OperationStatus.Completed ? 100 : checkpoint.PercentComplete,
-            Resources = await GetResourceUrlsAsync(type, checkpoint.ResourceIds, cancellationToken),
-            Status = status,
-            Type = type
-        };
-    }
+    public Task<OperationState<DicomOperation>> GetStateAsync(Guid operationId, CancellationToken cancellationToken = default)
+        => GetStateAsync(
+            operationId,
+            async (operation, state, checkpoint, token) =>
+            {
+                OperationStatus status = state.RuntimeStatus.ToOperationStatus();
+                return new OperationState<DicomOperation>
+                {
+                    CreatedTime = checkpoint.CreatedTime ?? state.CreatedTime,
+                    LastUpdatedTime = state.LastUpdatedTime,
+                    OperationId = operationId,
+                    PercentComplete = checkpoint.PercentComplete.HasValue && status == OperationStatus.Completed ? 100 : checkpoint.PercentComplete,
+                    Resources = await GetResourceUrlsAsync(operation, checkpoint.ResourceIds, cancellationToken),
+                    Status = status,
+                    Type = operation,
+                };
+            },
+            cancellationToken);
 
     /// <inheritdoc/>
-    public async Task<OperationCheckpointState<DicomOperation>> GetLastCheckpointAsync(Guid operationId, CancellationToken cancellationToken = default)
+    public async IAsyncEnumerable<OperationReference> FindOperationsAsync(OperationQueryCondition<DicomOperation> query, [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        cancellationToken.ThrowIfCancellationRequested();
-
-        DurableOrchestrationStatus state = await GetStatusInternalAsync(operationId);
+        EnsureArg.IsNotNull(query, nameof(query));
 
-        if (state == null)
+        var operations = new HashSet<DicomOperation>(query.Operations ?? Array.Empty<DicomOperation>());
+        var result = new OrchestrationStatusQueryResult();
+        do
         {
-            return null;
-        }
+            result = await _durableClient.ListInstancesAsync(
```

Comment:
Not sure about the performance of this query, if the operation status is not given, should we default to running status?

Reply:
Well, it's ultimately a query on the table in Azure Storage. We don't expose this API publicly, but it could be good to provide a more granular default

Reply:
I'll leave it without a default for now, but if we ever expose this publicly we'll need to consider the defaults